File: dataset/human/python/file_557.py
def input_ints():
    return list(map(int, input().split()))

def main():
    n = int(input())
    a = input_ints()
    c = 0
    for i in range(n):
        if a[i] < 0:
            c += 1
            a[i] *= -1
    ans = sum(a)
    if c % 2:
        ans -= 2 * min(a)
    print(ans)

if __name__ == '__main__':
    main()

File: dataset/human/c/file_557.c
#include <stdio.h>
#include <stdlib.h>

long long min(long long a,long long b){
  if(a < b){
    return a;
  }
  return b;
}

int main(void){
  long N;
  long long sum = 0,x = 0;
  long long ans = 10000000000000000;
  scanf("%ld",&N);

  long long a[N];

  for(int i = 0;i < N;i++){
    scanf("%lld",&a[i]);
    sum += a[i];
  }

  for(int i = 0;i < N-1;i++){
    x += a[i];
    ans = min(ans,llabs(sum-2*x));
  }

  printf("%lld\n",ans);

  return 0;
}
